inc/calendar.class.php

```diff
@@ -246,6 +246,24 @@ function cleanDBonPurge() {
    function isHoliday($date) {
       global $DB;
 
+      $end_month = ('mysql' === $DB->getDriver() ?
```

Comment:
To make addition of any new driver easier, it may be better to have following syntax:
```php
switch ($DB->getDriver()) {
   case 'mysql':
      $end_month = 'MONTH(' . $DB->quoteName('end_date') . ')';
      break;
   case 'pgsql':
      $end_month = 'EXTRACT(MONTH FROM ' . $DB->quoteName('end_date') . ')';
      break;
   default:
      throw new \RuntimeException(sprintf('Missing implementation for "%s" driver', $DB->getDriver()));
      break;
}
```
Indeed, with a switch syntax, adding a new case will be easier to merge, and any missing implementation will be easier to detect as it will throw an exception.

Reply:
In facts, we should have a more generic way than doing that at this place; but this is too much work to work on all functions alternatives a better way